### PR TITLE
net-mail/gnubiff-2.2.17: Fix 880267 (clang compilation), 930500 (LICENSE), 933241 (EAPI) 

### PR DESCRIPTION
**Pull request by UnixRonin**
net-mail/gnubiff: add 2.2.17-r3

Also adds a new LICENSE GPL-3+-with-openssl-exception

Closes: https://bugs.gentoo.org/880267
Closes: https://bugs.gentoo.org/930500
Closes: https://bugs.gentoo.org/933241

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by UnixRonin** [comment]
This is a clean resubmission of the original PR, after I made a complete and utter mess of squashing commits because I didn't understand how squashing worked, then proceeded to dig myself deeper from there.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+"
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-2.2.17-clang.patch"
+	eapply_user
+	eautoreconf
+}
```
Comment (AngryLoki) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:39]:
```suggestion
PATCHES=(
	"${FILESDIR}"/${PN}-2.2.17-clang.patch
)

src_prepare() {
	default
	eautoreconf
}
```

File: net-mail/gnubiff/files/gnubiff-2.2.17-clang.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+--- gnubiff-2.2.17/src/Makefile.am.orig	2016-06-12 04:18:12.000000000 -0400
```
Comment (AngryLoki) [inline net-mail/gnubiff/files/gnubiff-2.2.17-clang.patch:1]: Please apply `scrub-patch` (https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#clean-patch-howto)
and add short description mentioning #930500

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+#
+# This package depends upon deprecated gnome-base/libglade.  An
```
Comment (arthurzam) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:4]: wrong place for the comment, put it just above RDEPEND

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+# upstream but has been filed on this issue.
+# (https://sourceforge.net/p/gnubiff/bugs/67/)
+
+EAPI=8
+inherit autotools
```
Comment (arthurzam) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:9]:
```suggestion
EAPI=8

inherit autotools
```

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	local myeconfargs=(
+		--disable-gnome
+		$(use_enable debug)
+		$(use_enable nls)
+		$(use_enable fam)
+		$(use_with password)
+		$(use_with password password-string ${RANDOM}${RANDOM}${RANDOM}${RANDOM})
+	)
+	econf "${myeconfargs[@]}"
+}
```
Comment (arthurzam) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:58]:
```suggestion
src_configure() {
	local myeconfargs=(
		--disable-gnome # avoid deprecated gnome-panel-2.x
		$(use_enable debug)
		$(use_enable nls)
		$(use_enable fam)
		$(use_with password)
		$(use_with password password-string ${RANDOM}${RANDOM}${RANDOM}${RANDOM})
	)
	econf "${myeconfargs[@]}"
}
```

**Review by arthurzam — CHANGES_REQUESTED** [review]
please handle QA issue:
```
UnnecessaryManifest	unnecessary files in Manifest: [ gnubiff-2.2.17-clang.patch, gnubiff-2.2.17-configure.patch, gnubiff-2.2.17-fix-nls.patch, gnubiff-2.2.17-r2.ebuild, gnubiff-2.2.17-r3.ebuild, metadata.xml ]
```

I also think you drop patches used by the previous version, meaning premature removal

**Pull request closed without merging**